### PR TITLE
Small lint fixes

### PR DESCRIPTION
**Pull request by fregante**
- Updated shared config
- Enabled `--report-unused-disable-directives` and dropped the unused ones
- Replaced [`./` with `@/`](https://github.com/pixiebrix/pixiebrix-extension/issues/808) where they're equivalent, which [will be enforced soon](https://github.com/fregante/eslint-config-pixiebrix/commit/8dd0b53cd8d9e1d602ec7b348d45661e03e8e10d).
	`./` imports are still allowed in sub folders, like `src/devtools/something.ts` importing `./other.ts`

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -5,7 +5,6 @@
     "pixiebrix"
   ],
   "rules": {
-    "quotes": ["error", "double", { "avoidEscape": true }],
```
Comment (fregante) [inline .eslintrc:8]: Moved to config

File: .eslintrc [hunk 2]
```diff
@@ -28,6 +27,7 @@
     "@typescript-eslint/consistent-type-assertions": "warn",
     "@typescript-eslint/no-dynamic-delete": "warn",
     "@typescript-eslint/no-redeclare": "warn",
+    "@typescript-eslint/no-implicit-any-catch": "warn",
```
Comment (fregante) [inline .eslintrc:30]: Dropped from config, will also be dropped from here shortly together with:

- #779

File: package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "test:watch": "jest --watchAll",
     "test:selenium": "jest --config selenium/jest.config.js",
     "test:selenium:local": "BROWSERSTACK_USERNAME= jest --config selenium/jest.config.js",
-    "lint": "eslint src --ext js,jsx,ts,tsx",
+    "lint": "eslint src --ext js,jsx,ts,tsx --quiet --report-unused-disable-directives",
```
Comment (fregante) [inline package.json:10]: Warnings in this command are almost never useful due to the quantity. I will update the wiki to suggest `npm run lint -- --no-quiet` where you want to see everything.

This does not affect in-IDE warnings.

File: src/utils.ts [hunk 4]
```diff
@@ -299,13 +303,10 @@ export interface ReadProxy {
 export const noopProxy: ReadProxy = {
   toJS: identity,
   get: (value, prop) => {
-    if (
-      typeof value === "object" &&
-      Object.prototype.hasOwnProperty.call(value, prop)
-    ) {
+    if (isObject(value) && Object.prototype.hasOwnProperty.call(value, prop)) {
```
Comment (fregante) [inline src/utils.ts:306]: `isObject`:

- is a type-guard
- protects against null
- drops `as any` below

**Review by twschiller — APPROVED** [review]

**Pull request merged**